src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -17,20 +19,56 @@ public class Sections {
             cascade = {CascadeType.PERSIST, CascadeType.REMOVE})
     List<Section> sections = new ArrayList<>();
 
-    protected Sections() {
-    }
-
     public void add(Section section) {
+        if (!sections.isEmpty()) {
+            validateDuplicateSection(section);
+            Section connectionSection = findConnectableSection(section);
+            connectionSection.merge(section);
+        }
         sections.add(section);
     }
 
-    public List<Station> getStations() {
-        return sections.stream()
-                .flatMap(section -> Stream.of(section.getUpStation(), section.getDownStation()))
-                .collect(Collectors.toList());
+    private void validateDuplicateSection(Section section) {
+        if (getSectionStream(section).count() > 1) {
+            throw new IllegalArgumentException("상행역과 하행역이 이미 노선에 모두 등록되어있습니다.");
+        }
+    }
+
+    private Section findConnectableSection(Section section) {
+        return getSectionStream(section)
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("연결할 수 있는 구간이 없습니다."));
+    }
+
+    private Stream<Section> getSectionStream(Section section) {
+        return sections.stream().filter(s -> s.isContainAnyStation(section));
     }
 
-    public List<Section> getSections() {
-        return sections;
+    public List<Station> getStationsInOrder() {
+        Map<Station, Station> map = sections.stream()
+                .collect(Collectors.toMap(Section::getUpStation, Section::getDownStation));
+        Station station = findFirstUpStation();
+
+        List<Station> stations = new ArrayList<>();
+        while (map.get(station) != null) {
```

Comment:
```suggestion
        Station station = findFirstUpStation();

        List<Station> stations = new ArrayList<>();
        
        
        Map<Station, Station> map = sections.stream()
                .collect(Collectors.toMap(Section::getUpStation, Section::getDownStation));
        while (map.get(station) != null) {
```

`map` 
변수 선언과 사용하는곳 가깝게 놓는게 좋다고 생각해요 :)

Reply:
좋은 피드백 감사합니다!